homeassistant/components/soma/cover.py

```diff
@@ -20,40 +34,141 @@ async def async_setup_entry(
 ) -> None:
     """Set up the Soma cover platform."""
 
+    api = hass.data[DOMAIN][API]
     devices = hass.data[DOMAIN][DEVICES]
+    entities = []
+
+    for device in devices:
+        # Assume a shade device if the type is not present in the api response (Connect <2.2.6)
+        if "type" in device and device["type"].lower() == "tilt":
+            entities.append(SomaTilt(device, api))
+        else:
+            entities.append(SomaShade(device, api))
+
+    async_add_entities(entities, True)
+
 
-    async_add_entities(
-        [SomaCover(cover, hass.data[DOMAIN][API]) for cover in devices], True
+class SomaTilt(SomaEntity, CoverEntity):
+    """Representation of a Soma Tilt device."""
+
+    _attr_device_class = DEVICE_CLASS_BLIND
+    _attr_supported_features = (
+        SUPPORT_OPEN_TILT
+        | SUPPORT_CLOSE_TILT
+        | SUPPORT_STOP_TILT
+        | SUPPORT_SET_TILT_POSITION
     )
 
+    @property
+    def current_cover_tilt_position(self):
+        """Return the current cover tilt position."""
+        return self.current_position
+
+    @property
+    def is_closed(self):
+        """Return if the cover tilt is closed."""
+        return self.current_position == 0
+
+    def close_cover_tilt(self, **kwargs):
+        """Close the cover tilt."""
+        response = self.api.set_shade_position(self.device["mac"], 100)
+        if is_api_response_success(response):
+            self.set_position(0)
+        else:
+            raise HomeAssistantError(
+                f'Error while closing the cover ({self.name}): {response["msg"]}'
+            )
+
+    def open_cover_tilt(self, **kwargs):
+        """Open the cover tilt."""
+        response = self.api.set_shade_position(self.device["mac"], -100)
+        if is_api_response_success(response):
```

Comment:
If you reverse this condition and raise in the indent you can outdent the other side of the conditional and avoid the else 

This will save you a few lines in the other functions as well

Reply:
Good idea, fixed!